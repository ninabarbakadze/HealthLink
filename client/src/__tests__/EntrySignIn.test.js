import {render, screen} from '@testing-library/react';
import SignIn from '../components/signIn/signIn.tsx';
import userEvent from '@testing-library/user-event';
import {BrowserRouter as Router,Route,Switch} from 'react-router-dom';

describe('sign in component', () => {
  const mockSignIn = jest.fn(()=>{});
  const MockComponent = () => {
    return (
      <div>
      <p>mock</p>
      </div>
    )
  }

  test('should match the snapshot', () => {
    const {container} = render(
      <Router>
        <SignIn handleLogIn={mockSignIn}/>
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('email and password should be present on form',()=>{
    render(
      <Router>
        <SignIn handleLogIn={mockSignIn}/>
      </Router>
    );
    screen.getByPlaceholderText('Email Address');
    screen.getByPlaceholderText('Password');
  } );

  test('should not be able to attempt sign in if password or email are blank', ()=> {
    // const spy = jest.spyOn(SignInSide,'handleSubmit');
    render(
      <Router>
        <SignIn handleLogIn={()=>mockSignIn}/>
      </Router>
    );
    const email = screen.getByPlaceholderText('Email Address');
    const password = screen.getByPlaceholderText('Password');
    expect(email.value).toBe('');
    expect(password.value).toBe('');
    userEvent.click(screen.getByRole('button'));
    expect(mockSignIn).toHaveBeenCalledTimes(0);
  });

  test('sign up link takes you to sign up page',()=> {
    render(
      <Router>
        <SignIn handleLogIn={mockSignIn}/>
        <Switch>
        <Route exact path="/register" component={MockComponent}/>
        </Switch>
      </Router>
    );
    const signUp = screen.getByText("Don't have an account? Sign Up");
    userEvent.click(signUp);
    screen.getByText('mock');
  });
})