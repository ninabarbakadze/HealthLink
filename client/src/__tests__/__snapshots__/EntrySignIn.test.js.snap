// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign in component should match the snapshot 1`] = `
<div>
  <form>
    <h2>
      Sign In
    </h2>
    <label />
    <input
      name="email"
      placeholder="Email Address"
      required=""
      type="text"
      value=""
    />
    <label />
    <input
      name="password"
      placeholder="Password"
      required=""
      type="password"
      value=""
    />
    <button>
      Sign In
    </button>
    <span>
      <a
        href="/register"
      >
        Don't have an account? Sign Up
      </a>
    </span>
  </form>
</div>
`;
